src/main/ml-modules/root/server/lib/adhoc-create-lib.xqy

```diff
@@ -13,39 +14,69 @@ declare variable $form-fields-map :=
 	return $form-map
 ;
 
-declare function lib-adhoc-create:get-elementname($xpath as xs:string, $position as xs:string){
+declare variable $data-types-map :=
+let $dt-map := map:map()
+return $dt-map
+;
+
+declare function lib-adhoc-create:get-elementname($file-type as xs:string,$xpath as xs:string, $position as xs:string){
   let $tokens := fn:tokenize($xpath, "/")
   let $elementname :=
     if ($position eq "last") then
       $tokens[fn:last()]
-    else if($position eq "root") then $tokens[1] else $tokens[2]
+    else if($position eq "root") then
+	    if ($file-type = $const:FILE_TYPE_XML) then
+		  $tokens[1]
+		else if ( $file-type = $const:FILE_TYPE_JSON) then
+		  "/"
+		else ()
+	else
+		()
   return $elementname
 };
 
 declare function lib-adhoc-create:create-params($i){
   fn:concat('let $param', $i, ' := map:get($params, "id', $i, '")')
 };
 
-declare function lib-adhoc-create:create-evq($i, $xpath as xs:string){
- let $elementname := lib-adhoc-create:get-elementname($xpath, "last")
+declare function lib-adhoc-create:create-ewq($file-type as xs:string,$data-type as xs:string,$i, $xpath as xs:string) {
```

Comment:
Suggest we rename and or add comments at a minimum here. Text is being treated as a word query - meaning that it will match partial text strings such as "Mexico" matching "New Mexico". So we appear to be defaulting to word query everywhere, but of course non-text must be a value query (where "York" only matches "York" not "New York"). This returns different queries depending on type, and is then assigned to $evq (suggesting value queries). It's a little mixed up and not commented.